File: SerExtraESM/SerExtraESM.Web/Modules/Administration/AuditLog/AuditLogColumns.cs

namespace _Ext.Columns
{
    using Serenity.ComponentModel;
    using System;
    using System.ComponentModel;

    [ColumnsScript("_Ext.AuditLog")]
    [BasedOnRow(typeof(Entities.AuditLogRow))]
    public class AuditLogColumns
    {
        [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight, Hidden]
        public Int64 Id { get; set; }
        public String EntityTableName { get; set; }
        public AuditActionType ActionType { get; set; }
        public DateTime ActionDate { get; set; }
        public Int64 EntityId { get; set; }
        [Hidden]
        public String Changes { get; set; }
        [Width(200, Min = 150)]
        public Int64 UserId { get; set; }
        public String IpAddress { get; set; }
        [Hidden]
        public String SessionId { get; set; }
        [Hidden]
        public String RequestedURI { get; set; }
    }
}

File: SerExtraESM/SerExtraESM.Web/Modules/ServerTypes/_Ext/AuditLogColumns.ts
import { AuditLogActionTypeFormatter } from "@/Administration/AuditLog/AuditLogActionTypeFormatter";
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AuditActionType } from "./AuditActionType";
import { AuditLogRow } from "./AuditLogRow";

export interface AuditLogColumns {
    Id: Column<AuditLogRow>;
    EntityTableName: Column<AuditLogRow>;
    ActionType: Column<AuditLogRow>;
    ActionDate: Column<AuditLogRow>;
    EntityId: Column<AuditLogRow>;
    Changes: Column<AuditLogRow>;
    UserId: Column<AuditLogRow>;
    IpAddress: Column<AuditLogRow>;
    SessionId: Column<AuditLogRow>;
    RequestedURI: Column<AuditLogRow>;
}

export class AuditLogColumns extends ColumnsBase<AuditLogRow> {
    static readonly columnsKey = '_Ext.AuditLog';
    static readonly Fields = fieldsProxy<AuditLogColumns>();
}

[AuditLogActionTypeFormatter, AuditActionType]; // referenced types